From 547abd9c5a99cadda6453efc95ef965d6193c626 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 20 Dec 2021 06:36:48 +0200
Subject: [PATCH] Revert "[SW-63186] gaudi2: security: set axprot for mme"

As per alignment with LKD, MSTR_IF's OVRD register should not be used
to set to "USER level".

This is because sub-blocks which should be secure won’t be able to work,
as MSTR_IF's OVRD bit takes precedence over the engine's internal PROT
configuration.

This reverts commit 9cd8e85f06707168fbb982a3c3618fa119fd13b7.

Change-Id: I62d0ea50a2723d018b43deef31b7ee3d9f0f8c24
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index e7ddf0f..9c8a2c3 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -39,10 +39,6 @@
 #include <arc_farm_kdma_mstr_if_core_hbw_regs.h>
 #include <arc_farm_kdma_mstr_if_core_lbw_regs.h>
 #include <psoc_global_conf_regs.h>
-#include <dcore0_mme_sbte0_mstr_if_core_hbw_regs.h>
-#include <dcore0_mme_wb0_mstr_if_core_hbw_regs.h>
-#include <dcore0_mme_ctrl_mstr_if_core_hbw_regs.h>
-#include <dcore0_mme_ctrl_mstr_if_core_lbw_regs.h>
 
 /*
  * instances_num : Number of instances of RR to program.
@@ -203,7 +199,6 @@
 
 enum preboot_sec_axprot_regs {
 	AXPROT_ROT,
-	AXPROT_MME,
 };
 
 #define DCORE_DIST	0x200000
@@ -917,13 +912,6 @@
 #define ROT_RTR_INST_NUM	1
 #define ROT_INST_NUM		2
 #define ROT_INST_DIST		0x10000
-#define MME_SBTE_RTR_INST_NUM	5
-#define MME_SBTE_RTR_INST_DIST	0x8000
-#define MME_WB_RTR_INST_NUM	2
-#define MME_WB_RTR_INST_DIST	0x1000
-#define MME_INST_NUM		NUM_OF_DCORES
-#define MME_INST_DIST		DCORE_DIST
-#define MME_CRTL_RTR_INST_NUM	1
 
 static struct axprot_regs axprot_reg[] = {
 	[AXPROT_ROT] = {
@@ -933,14 +921,6 @@
 		.dcore_num = ROT_INST_NUM,
 		.inst_dist = 0,
 		.inst_num = ROT_RTR_INST_NUM,
-	},
-	[AXPROT_MME] = {
-		.base_reg = mmDCORE0_MME_SBTE0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
-		.value = AXPROT_USER_VAL,
-		.dcore_dist = MME_INST_DIST,
-		.dcore_num = MME_INST_NUM,
-		.inst_dist = MME_SBTE_RTR_INST_DIST,
-		.inst_num = MME_SBTE_RTR_INST_NUM,
 	},
 };
 
@@ -989,33 +969,6 @@
 	regmap_writel(PROT_OVR_USER, mmPSOC_GLOBAL_CONF_TRACE_AXPROT);
 }
 
-static void preboot_secure_mme(void)
-{
-	/* Configure mme to user level */
-
-	/* SBTE HBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_MME);
-
-	axprot_reg[AXPROT_MME].base_reg = mmDCORE0_MME_WB0_MSTR_IF_CORE_HBW_AXPROT_OVRD;
-	axprot_reg[AXPROT_MME].inst_dist = MME_WB_RTR_INST_DIST;
-	axprot_reg[AXPROT_MME].inst_num = MME_WB_RTR_INST_NUM;
-
-	/* WB HBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_MME);
-
-	axprot_reg[AXPROT_MME].base_reg = mmDCORE0_MME_CTRL_MSTR_IF_CORE_HBW_AXPROT_OVRD;
-	axprot_reg[AXPROT_MME].inst_dist = 0;
-	axprot_reg[AXPROT_MME].inst_num = MME_CRTL_RTR_INST_NUM;
-
-	/* CTRL HBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_MME);
-
-	axprot_reg[AXPROT_MME].base_reg = mmDCORE0_MME_CTRL_MSTR_IF_CORE_LBW_AXPROT_OVRD;
-
-	/* CTRL LBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_MME);
-}
-
 void preboot_security_init(void)
 {
 	preboot_secure_pcie();
@@ -1028,7 +981,6 @@
 		preboot_secure_pmmu();
 		preboot_secure_arc_farm();
 		preboot_secure_trace();
-		preboot_secure_mme();
 		preboot_secure_lbw();
 		preboot_secure_hbw();
 		preboot_puts("Security enabled.\n");
